src/main/java/persistence/sql/ddl/Entity.java

```diff
@@ -0,0 +1,80 @@
+package persistence.sql.ddl;
+
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.Transient;
+import persistence.sql.ddl.exception.IncorrectIdFieldException;
+import persistence.sql.ddl.exception.NotEntityException;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+public class Entity {
+    private String name;
+
+    private EntityIdField idField;
+
+    private List<EntityField> fields;
+
+    private Entity(String name, EntityIdField idField, List<EntityField> fields) {
+        this.name = name;
+        this.idField = idField;
+        this.fields = fields;
+    }
+
+    public static <T> Entity of(Class<T> clazz) {
+        if (clazz.getAnnotation(jakarta.persistence.Entity.class) == null) {
+            throw new NotEntityException();
+        }
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        return new Entity(
+                getName(clazz),
+                getIdField(declaredFields),
+                getFields(declaredFields)
+        );
+    }
+
+    private static <T> String getName(Class<T> clazz) {
+        Table table = clazz.getAnnotation(Table.class);
+
+        if (table == null) {
+            return clazz.getSimpleName();
+        }
+
+        return table.name();
+    }
+
+    private static EntityIdField getIdField(Field[] fields) {
+        List<Field> ids = Arrays.stream(fields)
+                .filter(it -> it.getAnnotation(Id.class) != null)
```

Comment:
```suggestion
                .filter(it -> it.isAnnotationPresent(Id.class))
```
의 기능을 제공하고 있어 사용해볼 수 있겠네요 😄